app/routes/endemics/date-of-visit-ms.js

```diff
@@ -45,254 +40,138 @@ const previousPageUrl = (latestVetVisitApplication, typeOfReview, previousClaims
-            }
-          )
-        }
-      ]
-    })
-  })
-
-  let data
-  const { error } = dateModel.validate(request.payload)
-  if (error) {
-    const errorSummary = []
-    const newError = addError(
-      error,
-      'visit-date',
-      'ifTheDateIsIncomplete',
-      'https://www.gov.uk/guidance/farmers-how-to-apply-for-funding-to-improve-animal-health-and-welfare#timing-of-reviews-and-follow-ups'
-    )
-    if (Object.keys(newError).length > 0 && newError.constructor === Object) { errorSummary.push(newError) }
-
-    const { latestVetVisitApplication, typeOfReview, previousClaims, typeOfLivestock } = session.getEndemicsClaim(request)
-
-    data = error
-      ? {
-          ...request.payload,
-          reviewOrFollowUpText,
-          errorSummary,
-          dateOfVisit: {
-            day: {
-              value: request.payload['visit-date-day'],
-              error: error.details.find(
-                (e) =>
-                  e.context.label === 'visit-date-day' ||
-                e.type.startsWith('dateOfVisit')
-              )
-            },
-            month: {
-              value: request.payload['visit-date-month'],
-              error: error.details.find(
-                (e) =>
-                  e.context.label === 'visit-date-month' ||
-                e.type.startsWith('dateOfVisit')
-              )
-            },
-            year: {
-              value: request.payload['visit-date-year'],
-              error: error.details.find(
-                (e) =>
-                  e.context.label === 'visit-date-year' ||
-                e.type.startsWith('dateOfVisit')
-              )
-            },
-            errorMessage: error.details.find((e) =>
-              e.context.label.startsWith('visit-date')
-            )
-              ? {
-                  text: error.details.find((e) =>
-                    e.context.label.startsWith('visit-date')
-                  ).message
-                }
-              : undefined
-          },
-          backLink: previousPageUrl(latestVetVisitApplication, typeOfReview, previousClaims, typeOfLivestock)
+const getOldWorldClaimFromApplication = (oldWorldApp, typeOfLivestock) =>
+  oldWorldApp && typeOfLivestock === oldWorldApp.data.whichReview
+    ? {
+        statusId: oldWorldApp.statusId,
+        data: {
+          claimType: oldWorldApp.data.whichReview,
+          dateOfVisit: oldWorldApp.data.visitDate
         }
-      : {}
+      }
+    : undefined
+
+const getInputErrors = (request, reviewOrFollowUpText) => {
+  const inputtedValues = {
+    day: request.payload[labels.day],
+    month: request.payload[labels.month],
+    year: request.payload[labels.year]
+  }
+
+  const inputsInError = {
+    day: false,
+    month: false,
+    year: false
+  }
+
+  if (inputtedValues.day === '' || Number.isNaN(Number(inputtedValues.day)) || inputtedValues.day > 31) {
```

Comment:
might we be better to use a Joi schema for this kind of validation?

Reply:
We had one initially but it was just massively over engineered. The way its been rewritten, we have full control over which input box is showing as red.